contracts/src/das/DasKeysetManager.sol

```diff
@@ -0,0 +1,63 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+// SPDX-License-Identifier: BUSL-1.1
+
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/AddressUpgradeable.sol";
+
+import "../libraries/DelegateCallAware.sol";
+
+/**
+ * @title Manage keysets for distributed Data Availability Service
+ * @notice Manages the valid keysets for distributed Data Availability Service
+ */
+contract DasKeysetManager is OwnableUpgradeable, DelegateCallAware {
+    using AddressUpgradeable for address;
+
+    uint256 public constant MAX_VALID_KEYSETS = 16;
+    bytes32[] private validKeysets;
+    mapping(bytes32 => uint256) private validKeysetIndex; // if keyset is at index in validKeysets, this is index+1
+    mapping(bytes32 => bytes) private contents;
+
+    function initialize() external initializer onlyDelegated {
+        __Ownable_init();
+    }
+
+    function isValidKeysetHash(bytes32 ksHash) external view returns (bool) {
+        return validKeysetIndex[ksHash] != 0;
+    }
+
+    function addKeyset(bytes calldata keyset) external onlyOwner {
+        require(validKeysets.length >= MAX_VALID_KEYSETS, "too many keysets");
+        bytes32 ksHash = keccak256(keyset);
+        if (this.isValidKeysetHash(ksHash)) {
+            return;
+        }
```

Comment:
Generally it's better to revert in a case like this since then tooling can just prevent the user from making the tx rather than spending the gas to post it